src/mjolnir/directededgebuilder.cc

```diff
@@ -48,8 +49,10 @@ DirectedEdgeBuilder::DirectedEdgeBuilder(const OSMWay& way,
 
   set_truck_route(way.truck_route());
 
-  // Set destination only to true if either destination only or no thru traffic is set
-  set_dest_only(way.destination_only() || way.no_thru_traffic());
+  // Set destination only to true if either destination only or no thru traffic is set or
+  // reclass_ferry is set to false.  Adding the reclass_ferry check fixes the issue
+  // https://github.com/valhalla/valhalla/issues/1889.
+  set_dest_only(way.destination_only() || way.no_thru_traffic() || reclass_ferry == false);
```

Comment:
this is not correct. this will set everything that hasnt been a reclassified ferry to be destination only! it should be like:

    reclass_ferry && (way.destination_only() || way.no_thru_traffic())